### PR TITLE
Support Inspector APIs needed for HOTUI.

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

File: src/io/flutter/inspector/InspectorService.java [hunk 1]
```diff
@@ -36,8 +57,127 @@
...
+    public int getOffset() {
+      return offset;
+    }
+
+    public VirtualFile getFile() {
+      return file;
+    }
+
+    public String getPath() {
+      return toSourceLocationUri(file.getPath());
```
Comment (pq) [inline src/io/flutter/inspector/InspectorService.java]: do we need to guard against a `null` file here?

File: src/io/flutter/inspector/InspectorService.java [hunk 2]
```diff
@@ -36,8 +57,127 @@
  * inspector code running in the IDE.
  */
 public class InspectorService implements Disposable {
+
+  public static class Location {
+
+    public Location(VirtualFile file, int line, int column, int offset) {
+      this.file = file;
```
Comment (pq) [inline src/io/flutter/inspector/InspectorService.java]: Can `file` be made non-nullable? (What would it mean to have a null one?)

File: src/io/flutter/inspector/InspectorService.java [hunk 3]
```diff
@@ -36,8 +57,127 @@
...
+      int nodeOffset = Math.max(0, Math.min(outline.getCodeOffset(), documentLength));
+      final int nodeEndOffset = Math.max(0, Math.min(outline.getCodeOffset() + outline.getCodeLength(), documentLength));
+
+      // The DartOutline will give us the offset of
+      // 'child: Foo.bar(...)'
+      // but we need the offset of 'bar(...)' for consistentency with the
+      // Flutter kernel transformer.
+      if (outline.getClassName() != null) {
+        final PsiFile psiFile = PsiDocumentManager.getInstance(project).getPsiFile(document);
+
```
Comment (pq) [inline src/io/flutter/inspector/InspectorService.java]: `psiFile` can be null; best to guard against that.

File: src/io/flutter/inspector/InspectorService.java [hunk 4]
```diff
@@ -577,6 +729,263 @@ private void skipIfDisposed(Runnable runnable) {
...
+        "}\n" +
+        "return false;\n";
+
+      return evaluateCustomApiHelper(command, new HashMap<>()).thenApplyAsync((instanceRef) -> {
+        return instanceRef != null && "true".equals(instanceRef.getValueAsString());
+      });
+    }
+
+    private CompletableFuture<InstanceRef> evaluateCustomApiHelper(String command, Map<String, String> scope) {
+      // Avoid running command if we interupted executing code as results will
```
Comment (pq) [inline src/io/flutter/inspector/InspectorService.java]: => interrupted

File: src/io/flutter/inspector/InspectorService.java [hunk 5]
```diff
@@ -577,6 +729,263 @@ private void skipIfDisposed(Runnable runnable) {
...
+      // be wierd. Repeatedly run the command until we hit idle.
```
Comment (pq) [inline src/io/flutter/inspector/InspectorService.java]: =>  weird

File: src/io/flutter/inspector/InspectorService.java (hunk 1 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java]: We don't. Added some missing @NotNull annotations to clarify.

File: src/io/flutter/inspector/InspectorService.java (hunk 2 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java]: done.

File: src/io/flutter/inspector/InspectorService.java (hunk 3 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java]: good point. done

File: src/io/flutter/inspector/InspectorService.java (hunk 4 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java]: done

File: src/io/flutter/inspector/InspectorService.java (hunk 5 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java]: done

**Pull request merged**